server/internal/core/application/covenantless.go

```diff
@@ -1404,6 +1424,13 @@ func (s *covenantlessService) startFinalization(roundEndTime time.Time) {
 	}
 	requests, boardingInputs, redeeemedNotes, musig2data := s.txRequests.pop(num)
 	notes = redeeemedNotes
+	vtxoKeys := make([]domain.VtxoKey, 0)
+	for _, req := range requests {
+		for _, in := range req.Inputs {
+			vtxoKeys = append(vtxoKeys, in.VtxoKey)
+		}
+	}
+	s.roundInputs.remove(vtxoKeys)
```

Comment:
Why removing the vtxo keys before signing the tree ? we should also prevent anybody to submit a redeem tx during the signing process (the keys should be removed when the round is done or failed) ?

Reply:
should be fixed